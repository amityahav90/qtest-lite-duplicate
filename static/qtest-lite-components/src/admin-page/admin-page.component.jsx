import React, { useEffect, useState } from 'react';
import Button from '@atlaskit/button';
import './admin-page.styles.scss';
import { useLogHook, withLogPath } from '../shared/log-service';
import i18n from '../shared/localization/i18n';
import { withErrorBoundaryRoot } from '../shared/error-boundary-root';
import { ExternalCommunicationService } from '../shared/external-communication';
import Spinner from '@atlaskit/spinner';

function AdminPageInternal() {
	const log = useLogHook();
	const [appInitialized, setAppInitialized] = useState(false);
	const externalCommunicationService = new ExternalCommunicationService();
	const [initializationDisabled, setInitializationDisabled] = useState(false);
	const [internalVersion, setInternalVersion] = useState('');

	useEffect(() => {
		externalCommunicationService
			.getInternalSettings()
			.then((internalSettings) => {
				setAppInitialized(internalSettings.initializedApp);
			})
			.catch((e) => {});

		externalCommunicationService
			.getApplicationInternalVersion()
			.then((version) => {
				setInternalVersion(version);
			})
			.catch((e) => {});
	}, []);

	const initializeApp = () => {
		setInitializationDisabled(true);

		log.debug('Initializing the storage service.');

		// Calling the app initialization function
		externalCommunicationService
			.initializeApp()
			.then(() => {
				log.info('App initialization has finished successfully.');
				setAppInitialized(true);
			})
			.catch((e) => {
				log.error('Failed to complete the app initialization' + e);
			});
	};

	const resetInitializedApp = () => {
		externalCommunicationService
			.resetInitializeApp()
			.then(() => {
				log.debug('App initialized flag was reset successfully');
				setAppInitialized(false);
				setInitializationDisabled(false);
			})
			.catch((e) => {
				log.error('Failed to reset the app initialization flag.', e);
			});
	};

	return (
		<div className="admin-page">
			<div className="admin-page-actions">
				{internalVersion && (
					<div className="admin-page-internal-version">{internalVersion}</div>
				)}
				<div className="admin-page-initialize">
					{!appInitialized ? (
						<div className="app-not-initialized">
							<Button
								id="initialize-app-btn"
								appearance="primary"
								isDisabled={initializationDisabled}
								onClick={() => initializeApp()}
							>
								{i18n.t('ADMIN_PAGE_INITIALIZE_APP_BTN')}
							</Button>
							{initializationDisabled && (
								<div className="initializing-app-spinner">
									<Spinner size="large" />
								</div>
							)}
						</div>
					) : (
						<div className="app-initialized">
							<div className="app-initialized-txt">App is already initialized.</div>
							<Button id="reset-btn" appearance="primary" onClick={() => resetInitializedApp()}>
								Reset
							</Button>
						</div>
					)}
				</div>
			</div>
		</div>
	);
}

export const AdminPage = withLogPath(withErrorBoundaryRoot(AdminPageInternal), 'AdminPage');
