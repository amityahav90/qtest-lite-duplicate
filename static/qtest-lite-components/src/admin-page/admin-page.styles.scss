.admin-page {
	display: flex;

	.admin-page-actions {
		display: flex;
		padding: 15px;
		flex-direction: column;
		margin-bottom: 20px;

		.admin-page-internal-version {
			display: flex;
			align-items: center;
			font-size: 12px;
		}

		.admin-page-initialize {
			display: flex;
			justify-content: center;
			align-items: center;

			.app-not-initialized {
				display: flex;
				flex-direction: column;

				.initializing-app-spinner {
					margin-top: 20px;
					align-self: center;
				}
			}

			.app-initialized {
				display: flex;
				flex-direction: column;

				.app-initialized-txt {
					color: #42526e;
					margin-bottom: 8px;
				}
			}
		}
	}
}
