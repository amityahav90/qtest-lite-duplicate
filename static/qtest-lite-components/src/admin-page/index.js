import React from 'react';
import ReactDOM from 'react-dom';
import { AdminPage } from './admin-page.component';

import '@atlaskit/css-reset';

ReactDOM.render(
	<React.StrictMode>
		<AdminPage />
	</React.StrictMode>,
	document.getElementById('root')
);
