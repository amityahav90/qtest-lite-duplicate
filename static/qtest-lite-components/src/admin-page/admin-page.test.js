import React from 'react';
import renderer from 'react-test-renderer';
import { AdminPage } from './admin-page.component';

/**
 * Set of unit tests performed on the AdminPage component
 */
describe('Test AdminPage component', () => {
	let component;

	/**
	 * Runs before all tests are executed and prepare the testing environment.
	 */
	beforeAll(() => {
		jest.mock('@forge/bridge');
		component = renderer.create(<AdminPage />);
	});

	/**
	 * Snapshot test of the AdminPage component.
	 */
	test('App is rendered correctly', async () => {
		expect(component.toJSON()).toMatchSnapshot();
	});
});
