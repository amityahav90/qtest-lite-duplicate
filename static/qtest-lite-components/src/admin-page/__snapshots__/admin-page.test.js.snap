// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AdminPage component App is rendered correctly 1`] = `
<div
  className="admin-page"
>
  <div
    className="admin-page-actions"
  >
    <div
      className="admin-page-initialize"
    >
      <div
        className="app-not-initialized"
      >
        <button
          className="css-7bs0bd-ButtonBase"
          disabled={false}
          id="initialize-app-btn"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="css-7no60z-ButtonBase"
          >
            Initialize App
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
